src/App/App.js

```diff
@@ -13,25 +13,32 @@ export class App extends Component {
     super(props);
     this.state = {
       user: {},
-      isOnHomePage: true
+      isOnHomePage: true,
+      error: ''
     }
   }
 
   updateCurrentUser = ({ user }) => {
     this.setState({ user }, () => {this.props.history.push('/')})
   }
 
-  handleLoginSubmit = (userInfo) => {
-    ApiCalls.postUserLogin(userInfo)
-    .then(data => this.updateCurrentUser(data))
+  handleLoginSubmit = async (userInfo) => {
+    const loginResult = await ApiCalls.postUserLogin(userInfo)
+    if (loginResult.error) {
+      this.setState({error: loginResult.error})
```

Comment:
Nice error handling here. I think showing errors in state might be a good way to go for displaying the error to the user.